### PR TITLE
Addon-info: allow duplicate displayNames

### PR DESCRIPTION
**Pull request by Hypnosphi**
Issue: #1814 
When using HOCs, different component types may have the same names. #1607 fixed it for `name` property, but duplicate `diplayName`s still lead to "same key" warnings.

## What I did
Added indices to propTable div keys

## How to test
`yarn start` & open http://localhost:9010/?selectedKind=GitHub%20issues&selectedStory=%231814

### CODE REVIEW

File: addons/info/src/components/Story.js [hunk 1]
```diff
@@ -324,14 +326,13 @@ export default class Story extends React.Component {
...
 
     const { maxPropObjectKeys, maxPropArrayLength, maxPropStringLength } = this.props;
-    const propTables = array.map(type => (
-      <div key={type.displayName || type.name}>
-        <h2 style={this.state.stylesheet.propTableHead}>
-          "{type.displayName || type.name}" Component
-        </h2>
+    const propTables = array.map((type, i) => (
+      // eslint-disable-next-line react/no-array-index-key
+      <div key={`${getName(type)}_${i}`}>
```
Comment (Hypnosphi) [inline addons/info/src/components/Story.js]: I think it's OK to use indices in this case

File: examples/cra-kitchen-sink/src/stories/issues.stories.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { storiesOf } from '@storybook/react';
+import { withInfo } from '@storybook/addon-info';
+
+const hoc = WrapComponent => ({ ...props }) => <WrapComponent {...props} />;
+
+const Input = hoc(() => <input type="text" />);
+
+const TextArea = hoc(({ children }) => <textarea>{children}</textarea>);
+
+storiesOf('GitHub issues', module).add(
```
Comment (danielduan) [inline examples/cra-kitchen-sink/src/stories/issues.stories.js]: Naming it github issues isn't very helpful. Maybe `Addon Info.HOC` and `withInfo('Allow Duplicate DisplayNames for HOC #1814')`?

Comment (Hypnosphi) [reply examples/cra-kitchen-sink/src/stories/issues.stories.js]: I wanted to introduce a single namespace for reproductions

Comment (danielduan) [reply examples/cra-kitchen-sink/src/stories/issues.stories.js]: I understand, but ultimately, I still think this is a `Addon Info` thing. If we have 20 different `Github` issues and I'm making a change to `Addon Info`, I wouldn't think to check all the `Github` ones for regressions you know?

Maybe `Addon Info.Github Issues` and that description I had earlier?

**Review by danielduan — APPROVED** [review]

**Pull request merged**